src/main/java/org/wise/portal/service/peergrouping/logic/impl/DifferentIdeasLogicServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+package org.wise.portal.service.peergrouping.logic.impl;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.SetUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.wise.portal.dao.annotation.wise5.AnnotationDao;
+import org.wise.portal.domain.group.Group;
+import org.wise.portal.domain.peergrouping.PeerGrouping;
+import org.wise.portal.domain.peergrouping.logic.DifferentIdeasLogic;
+import org.wise.portal.domain.workgroup.Workgroup;
+import org.wise.portal.service.peergroup.impl.WorkgroupWithDifferentIdeas;
+import org.wise.vle.domain.annotation.wise5.Annotation;
+
+@Service
+public class DifferentIdeasLogicServiceImpl extends PeerGroupLogicServiceImpl {
+
+  @Autowired
+  private AnnotationDao<Annotation> annotationDao;
+
+  boolean canCreatePeerGroup(Workgroup workgroup, Set<Workgroup> workgroupsNotInPeerGroup,
+      PeerGrouping peerGrouping) {
+    List<Annotation> annotations = getIdeaDetectedAnnotations(peerGrouping, workgroup.getPeriod());
+    return hasWorkgroupAnnotation(workgroup, annotations)
+        && hasEnoughUnpairedMembersIdeasAnnotation(peerGrouping, annotations,
+            workgroupsNotInPeerGroup);
+  }
+
+  private List<Annotation> getIdeaDetectedAnnotations(PeerGrouping peerGrouping, Group period) {
+    DifferentIdeasLogic logic = new DifferentIdeasLogic(peerGrouping.getLogic());
+    List<Annotation> annotations = annotationDao
+        .getAnnotations(peerGrouping.getRun(), logic.getNodeId(), logic.getComponentId()).stream()
+        .filter(annotation -> annotation.getPeriod().equals(period)
+            && annotation.getType().equals("autoScore"))
+        .collect(Collectors.toList());
```

Comment:
Should we call annotationDao.getAnnotationsByParams() which can filter by period id and type?